pal/minigame/wechat_mini_program.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ Copyright (c) 2022-2023 Xiamen Yaji Software Co., Ltd.
+
+ https://www.cocos.com/
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights to
+ use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ of the Software, and to permit persons to whom the Software is furnished to do so,
+ subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+
+import { IMiniGame, SystemInfo } from 'pal/minigame';
+import { Orientation } from '../screen-adapter/enum-type';
+import { cloneObject, createInnerAudioContextPolyfill, versionCompare } from '../utils';
+
+declare let wx: any;
+
+// @ts-expect-error can't init minigame when it's declared
+const minigame: IMiniGame = {};
+cloneObject(minigame, wx);
+
+// #region platform related
+minigame.wx = {};
+minigame.wx.onKeyDown = wx.onKeyDown?.bind(wx);
+minigame.wx.onKeyUp = wx.onKeyUp?.bind(wx);
+minigame.wx.onMouseDown = wx.onMouseDown?.bind(wx);
+minigame.wx.onMouseMove = wx.onMouseMove?.bind(wx);
+minigame.wx.onMouseUp = wx.onMouseUp?.bind(wx);
+minigame.wx.onWheel = wx.onWheel?.bind(wx);
+// #endregion platform related
+
+// #region SystemInfo
+let _cachedSystemInfo: SystemInfo = wx.getSystemInfoSync();
+// @ts-expect-error TODO: move into minigame.d.ts
+minigame.testAndUpdateSystemInfoCache = function (testAmount: number, testInterval: number) {
```

Comment:
It is better to fix eslint issue.

Reply:
I DO NOT modify this file, just copy from wechat game. so I think It’s better to be consistent